.github/workflows/release.yml

```diff
@@ -7,14 +7,18 @@ on:
 
 jobs:
   goreleaser:
+    permissions:
+      id-token: write
+      contents: write
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - run: git fetch --prune --unshallow
-      - uses: actions/setup-go@v1
+      - uses: actions/setup-go@v2
         with:
           go-version: 1.17.x
-      - uses: goreleaser/goreleaser-action@v1
+      - uses: sigstore/cosign-installer@v1.4.1
+      - uses: goreleaser/goreleaser-action@v2
```

Comment:
Could we add a step after goreleaser that runs `cosign verify <checksum>` to check that this worked as expected?

Reply:
Question: Is cosign validating this already? I like the idea of validating the signature after signature creation, but I am not sure if validating makes sense if cosign is validating it already during creation.

Reply:
nice let's do this.